sdk/core/Azure.Core/src/Shared/NextLinkOperationImplementation.cs

```diff
@@ -122,7 +126,7 @@ internal class NextLinkOperationImplementation : IOperation
                 finalStateVia = OperationFinalStateVia.Location;
             }
 
-            string headerSourceStr = GetContentFromRehydrationToken(lroDetails, "headerSource");
+            string headerSourceStr = GetContentFromRehydrationToken(lroDetails, "headerSource")!;
```

Comment:
If we're sure that the outcome here is not null, why is `GetContentFromRehydrationToken` marked as nullable?